app/services/company/delete_company.rb

```diff
@@ -0,0 +1,26 @@
+class DeleteCompany
+  include Callable
+
+  def initialize(company_id)
+    @company = Company.find(company_id)
+  end
+
+  def call
+    destroy if valid?
+  end
+
+  private
+
+  def valid?
+    if @company.nil?
+      errors.add(:base, 'Company not found')
+      return false
+    else
+      return true
+    end
+  end
```

Comment:
Aqui poderia fazer:

 def valid?
   return true  if @company.present?
   
  errors.add(:base, 'Company not found')
  false      
end

Reply:
- [x] implementada essa forma mais elegante no método de validação